agda/BinaryRandomAccessList.agda

```diff
@@ -37,21 +40,68 @@ module BinaryRandomAccessList where
   open import Structure.Bin
   open import Container.LeafBinaryTree
 
-  RAL : Set → Set
-  RAL A = DBin Tree
+  RAL-g : Set → ℕ → Set
+  RAL-g A n = DBin-g Tree 
                Container.LeafBinaryTree.toVec Container.LeafBinaryTree.fromVec
-               Container.LeafBinaryTree.iso-to-from Container.LeafBinaryTree.iso-from-to A
+               Container.LeafBinaryTree.iso-to-from Container.LeafBinaryTree.iso-from-to A n
 
+  RAL : Set → Set
+  RAL A = RAL-g A 0
 
   -- TODO: implement all the operations on the datastructure:
   --   cf. "Numerical representation as higher-order nested datatypes", Fig. 3, p.13 for the interface
   --   cf. Okasaki, Sec. 9.2.1 and 10.1.2 for the implementation
 
+  -- Returns an empty structure
   empty : ∀ {A} → RAL A
-  empty = {!!}
-
+  empty = ϵ
+
+  private
+    -- Appends a new Tree in a structure of trees
+    consTree : ∀ {A n} → Tree A n → RAL-g A n → RAL-g A n
```

Comment:
Add a lemma relating the size of the underlying containers: something like

    forget (consTree l t) = BintoNat (size l) +Bin forget t

(Catherine will explicit need this)